server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -628,10 +631,7 @@ private void parseValueAddFields(ParseContext context, String value, String fiel
             }
 
             if (fieldType().hasDocValues()) {
-                if (context.doc().getField(fieldType().name()) == null || !context.doc().getFields(fieldType().name()).equals(field)) {
-                    if (fieldName.equals(fieldType().name())) {
-                        context.doc().add(new SortedSetDocValuesField(fieldType().name(), binaryValue));
-                    }
+                if (!Objects.equals(context.doc().getField(fieldType().name()), field)) {
```

Comment:
It needs import please:

```
import java.util.Objects;
```

Reply:
Fixed.
(I clicked the Suggestion Merge button on GitHub, but it did not work..)

Reply:
It seems like it did? https://github.com/opensearch-project/OpenSearch/pull/7855/files#diff-8a5849247d51dba2b09b6505e299c618dbd3ca9b38d3ee1a30b07941c7bd7a9bR51

Reply:
Oh, it was not the suggestion since the import section didn't change, sorry about the confusion

Reply:
Thank you for checking it.
The "Suggestion Merge button" is GitHub feature. I clicked it without adding import...